src/test/scala/chiselTests/util/SizeOfSpec.scala

```diff
@@ -0,0 +1,48 @@
+package chiselTests.util
+
+import chisel3._
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage}
+import chisel3.testers.BasicTester
+import chisel3.util.circt.SizeOf
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.io.Source
+
+class SizeOfTop extends Module {
+  val io = IO(new Bundle {
+    val w = Input(UInt(65.W))
+    val x = Input(UInt(66.W))
+    val outw = UInt(32.W)
+    val outx = UInt(32.W)
+  })
+  io.outw := SizeOf(io.w)
+  io.outx := SizeOf(io.x)
+}
+
+/** A test for intrinsics.  Since chisel is producing intrinsics as tagged
+  * extmodules (for now), we explicitly test the chirrtl and annotations rather
+  * than the processed firrtl or verilog.  It is worth noting that annotations
+  * are implemented (for now) in a way which makes the output valid for all
+  * firrtl compilers, hence we write a localized, not end-to-end test
+  */
+class SizeOfSpec extends AnyFlatSpec with Matchers {
+  it should "Should work for types" in {
+    val fir = ChiselStage.emitChirrtl(new SizeOfTop)
+    println(fir)
+    val a1 = """extmodule SizeOf_4""".r
+    (fir should include).regex(a1)
+    val b1 = """defname = SizeOf_4""".r
+    (fir should include).regex(b1)
+    val a2 = """extmodule SizeOf_e""".r
+    (fir should include).regex(a2)
+    val b2 = """defname = SizeOf_e""".r
+    (fir should include).regex(b2)
+
+    // The second elaboration uses a unique name since the Builder is reused (?)
+    val anno = ChiselStage.emitAnnotations(new SizeOfTop)
+    val c = """Intrinsic\(~SizeOfTop\|SizeOf.*,circt.sizeof\)"""
+    (anno should include).regex(c)
```

Comment:
This can use patten matching to clean this up somewhat. Consider:

```scala
  ((new chisel3.stage.ChiselStage)
    .execute(
      args = Array("--no-run-firrtl"),
      annotations = Seq(chisel3.stage.ChiselGeneratorAnnotation(() => new SizeOfTop))
    )
    .mkString("\n") should include).regex(c)
```

Alternatively, if you make the Intrinsic annotation package private `private[chisel3]` and place this test in the `chisel3` package, you can then test this more directly without using strings. I.e., the test is in the `chisel3` package, has access to the annotation, and can then pattern match on it directly.